gems/smithy-cbor/lib/smithy-cbor.rb

```diff
@@ -40,52 +33,65 @@ def initialize(tag, value)
       attr_accessor :value
     end
 
-    # Generic CBOR error, super class for specific encode/decode related errors.
-    class Error < StandardError; end
+    # Raised when a CBOR build error occurs.
+    class BuildError < StandardError; end
+
+    # Raised when a CBOR parsing error occurs.
+    class ParseError < StandardError; end
 
-    # Malformed buffer, expected more bytes
-    class OutOfBytesError < Error
-      def initialize(requested_bytes, left)
-        super("Out of bytes. Trying to read #{requested_bytes} bytes but buffer contains only #{left}")
+    class << self
+      # @param [Symbol, Class] engine
+      #   Must be one of the following values:
+      #
+      #   * :smithy
```

Comment:
I'm not crazy about `:smithy` or `SmithyEngine` naming but I see that's best we can name this. 😅